build-conflicts.php

```diff
@@ -187,79 +123,64 @@ static function (array $components): array {
-        runInPath(
-            static function () use ($execute): void {
-                $execute('composer validate');
-            },
-            dirname($composerJsonPath)
-        );
+    $validateComposerJson = static function (string $composerJsonPath): void {
+        Shell\execute('php', ['composer.phar', 'validate'], Filesystem\get_directory($composerJsonPath));
     };
 
     $copyGeneratedComposerJson = static function (
         string $sourceComposerJsonPath,
         string $targetComposerJsonPath
-    ) use ($execute): void {
-        $execute(sprintf(
-            'cp %s %s',
-            escapeshellarg($sourceComposerJsonPath),
-            escapeshellarg($targetComposerJsonPath)
-        ));
+    ): void {
+        Shell\execute('cp', [$sourceComposerJsonPath, $targetComposerJsonPath]);
     };
 
-    $commitComposerJson = static function (string $composerJsonPath) use ($execute): void {
-        $parseHead =
-            /** @psalm-return non-empty-list<string> */
-            static function () use ($execute): array {
-                return $execute('git rev-parse HEAD');
-            };
-        $originalHash = runInPath(
-            $parseHead,
-            dirname($composerJsonPath) . '/../security-advisories'
+    $commitComposerJson = static function (string $composerJsonPath): void {
+        $originalHash = Shell\execute(
+            'git',
+            ['rev-parse', 'HEAD'],
+            Filesystem\get_directory($composerJsonPath) . '/../security-advisories'
         );
+        $originalHash = Str\trim($originalHash);
 
-        runInPath(
-            static function () use ($composerJsonPath, $originalHash, $execute): void {
-                $execute('git add ' . escapeshellarg(realpath($composerJsonPath)));
-
-                $message  = sprintf(
-                    'Committing generated "composer.json" file as per "%s"',
-                    (new DateTime('now', new DateTimeZone('UTC')))->format(DateTime::W3C)
-                );
-                $message .= "\n" . sprintf(
-                    'Original commit: "%s"',
-                    'https://github.com/FriendsOfPHP/security-advisories/commit/' . $originalHash[0]
-                );
-
-                $execute('git diff-index --quiet HEAD || git commit -m ' . escapeshellarg($message));
-            },
-            dirname($composerJsonPath)
+        $workingDirectory = Filesystem\get_directory($composerJsonPath);
+        Shell\execute('git', ['add', (string) Filesystem\canonicalize($composerJsonPath)], $workingDirectory);
+
+        $message  = Str\format(
+            'Committing generated "composer.json" file as per "%s"',
+            (new DateTime('now', new DateTimeZone('UTC')))->format(DateTime::W3C)
+        );
+        $message .= "\n" . Str\format(
+            'Original commit: "%s"',
+            'https://github.com/FriendsOfPHP/security-advisories/commit/' . $originalHash[0]
```

Comment:
*bug*: $originalHash is not an array anymore, instead it's a string, so now it's writing the first character instead of the first line 😅